### DISCUSSION TITLE
LXD import or/and integration of VirtualMachines

### DISCUSSION THREAD

**Discussion by erik78se** [netbox-community/netbox#15018] [General]
**My setup:** I have a few LXD instances which are represented by Cluster objects in Netbox (the machines are Devices).

**What I want to achieve:** I want to inventory the instances running on the LXD hosts as "VirtualMachines" and keep the inventory automatically in sync to whats on the LXD hosts.

The challenge is that LXD hosts runs a dynamic set of containers. Netbox will somehow need to be updated automatically, for example when instances changes state, gets added or removed.

**Questions in relation to my scenario:**

A) Is there some LXD integration known today for Nextbox with LXD that covers my scenario?

B) If not, is there a way to automatic-periodically import lxc container metadata into Netbox, lets say from files in S3, these files would then be  periodically sent from my running LXD instaces?

C) Is there some other way to cover my scenario that I should pursue?

Many thanx for your input.

**Comment by candlerb** [comment]
Please see previous reply at https://github.com/netbox-community/netbox/discussions/10789#discussioncomment-8345483
